test/mason/publish-tests/publish.good

```diff
@@ -0,0 +1,12 @@
+Created new library project: test_dir
```

Comment:
How about calling this directory `mason-publish` or just `publish` to match what other test directories do in `mason/` dir?